cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -198,96 +199,87 @@ public ConnectOperatorMetricsHolder metrics()   {
      * identified by {@code KafkaConnector.metadata.labels[strimzi.io/cluster]}.</li>
      * <li>The {@code KafkaConnector} status is updated with the result.</li>
      * </ul>
-     * @param connectOperator The operator for {@code KafkaConnect}.
-     * @param watchNamespaceOrWildcard The namespace to watch.
-     * @param selectorLabels Selector labels for filtering the custom resources
+     *
+     * @param namespace Namespace where to watch.
+     * @param onClose Callback called when the watch is closed.
      *
      * @return A future which completes when the watch has been set up.
      */
-    public static Future<Watch> createConnectorWatch(AbstractConnectOperator<KubernetesClient, KafkaConnect, KafkaConnectList, Resource<KafkaConnect>, KafkaConnectSpec, KafkaConnectStatus> connectOperator,
-                                                     String watchNamespaceOrWildcard, Labels selectorLabels) {
+    public Future<Watch> createConnectorWatch(String namespace, Consumer<WatcherException> onClose) {
+        Labels selectorLabels =  config.getCustomResourceSelector();
```

Comment:
Ah right, I just wanted to avoid the extra parameter to have the method signature as close to the createWatch() method as possible, but you are right, retaining the extra state it probably not worth it. Since the config is already passed to the constructor maybe it would satisfy us both to just store the selectorLabels in the object

Reply:
Removed